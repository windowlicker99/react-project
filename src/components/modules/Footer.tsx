import React, { FC } from 'react';
import { useSelector } from 'react-redux';
import { Pagination } from '@components/common/Pagination';
import { currentTab, elementAmount, showedAmount } from '@/store/selectors/pagination';

export const Footer: FC = () => {
  const tab = useSelector(currentTab);
  const totalElements = useSelector(elementAmount);
  const elements = useSelector(showedAmount);

  return (
    <footer className="footer">
      <div className="footer-inner ">
        <span className="footer-text">
          Showing {elements} of {totalElements} {tab}
        </span>
        <Pagination />
      </div>
    </footer>
  );
};
