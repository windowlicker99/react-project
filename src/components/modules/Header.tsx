import React, { FC } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router';
import { Search } from '@components/common/form-controls/Search';
import { HeaderDropdown } from '@components/common/HeaderDropdown';
import { headerSubtitles, showHeaderArrow } from '@/store/selectors/header';
import { setHeader } from '@/store/slices/headerSlice';
import { CLIENT_PATHS } from '@/constants/paths';

interface IHeaderProps {
  title: string;
}

export const Header: FC<IHeaderProps> = ({ title }) => {
  const dispatch = useDispatch();
  const history = useHistory();
  const showArrow = useSelector(showHeaderArrow);
  const subtitles = useSelector(headerSubtitles);

  const backBtnClick = () => {
    switch (title) {
      case 'Architecture':
        history.push(`${CLIENT_PATHS.sidebar}${CLIENT_PATHS.vehicle}`);
        break;
      default:
        history.goBack();
        break;
    }
    dispatch(setHeader([]));
  };

  return (
    <header className="header">
      <div className="container">
        <nav className="header-nav">
          <div className="header-inner">
            <div className="header-toggler">
              {showArrow ? (
                <span className="icon-arrow-left" onClick={backBtnClick} aria-hidden="true" />
              ) : (
                <span className="icon-menu" />
              )}
            </div>

            <div className="header-title">
              {title}

              {showArrow &&
                subtitles.map((subtitle) => (
                  <span className="header-subtitle" key={subtitle}>
                    {subtitle}
                  </span>
                ))}
            </div>
          </div>
          <div className="header-controls">
            <div className="mr-15">
              <Search formPlaceholder="Search" />
            </div>
            <HeaderDropdown />
          </div>
        </nav>
      </div>
    </header>
  );
};
