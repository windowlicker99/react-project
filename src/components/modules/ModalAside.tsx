import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

interface IModalAsideProps {
  isOpen: boolean;
  header?: ReactNode;
  footer?: ReactNode;
}

export const ModalAside: FC<IModalAsideProps> = ({ isOpen, header, footer, children }) => (
  <div className={classNames('modal modal-aside', { show: isOpen })}>
    {/* <!-- add class show --> */}
    <div className="modal-content">
      {header && <div className="modal-header">{header}</div>}
      <div className="modal-body">{children}</div>
      {footer && <div className="modal-footer">{footer}</div>}
    </div>
  </div>
);
