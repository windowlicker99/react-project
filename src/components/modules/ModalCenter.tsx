import React, { FC } from 'react';
import classNames from 'classnames';
import { Btn } from '@components/common/form-controls/Btn';

interface IModalCenter {
  title: string;
  subtitle?: string;
  isOpen: boolean;
  disabledApplyBtn?: boolean;
  onClose: () => void;
  onApply: () => void;
}

export const ModalCenter: FC<IModalCenter> = ({
  title,
  subtitle = '',
  disabledApplyBtn = false,
  isOpen,
  onClose,
  onApply,
  children,
}) => (
  <div className={classNames('modal', { show: isOpen })}>
    <div className="modal-content">
      <div className="modal-header">
        <div className="modal-heading">
          <div className="modal-header-content">
            <span className="modal-title">{title}</span>
            <span className="modal-subtitle">{subtitle}</span>
          </div>
          {title !== 'Questionnaire' && (
            <button type="button" className="btn btn-xs-secondary" onClick={onClose}>
              <span className="icon-close" />
            </button>
          )}
        </div>
      </div>
      <div className="modal-body">{children}</div>
      <div className="modal-footer">
        <div className="btn-group-stretch">
          <Btn btnStatus="btn-lg-outline mr-15" btnIcon="" btnText="Cancel" onClick={onClose} />
          <Btn btnStatus="btn-lg-primary" disabledBtn={disabledApplyBtn} btnIcon="" btnText="Apply" onClick={onApply} />
        </div>
      </div>
    </div>
  </div>
);
