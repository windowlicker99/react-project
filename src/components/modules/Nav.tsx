import React, { FC } from 'react';

export const Nav: FC = () => (
  <nav className="nav">
    <ul className="nav-list">
      <li className="nav-item">
        <a href="/" className="nav-link">
          Item 1
        </a>
      </li>
      <li className="nav-item">
        <a href="/" className="nav-link">
          Item 2
        </a>
      </li>
      <li className="nav-item">
        <a href="/" className="nav-link">
          Item 3
        </a>
      </li>
    </ul>
  </nav>
);
