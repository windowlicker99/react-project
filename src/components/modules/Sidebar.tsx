import React, { FC } from 'react';
import { Switch, NavLink, useRouteMatch } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { Logo } from '@components/common/Logo';
import { sidebarTabsList } from '@constants/tabsLists';
import { PageContentLayout } from '@components/page-contents/PageContentLayout';
import { SecurityPage } from '@components/pages/SecurityPage';
import { PrivateRoute } from '@routing/PrivateRoute';
import { setHeader } from '@store/slices/headerSlice';
import { CLIENT_PATHS } from '@constants/paths';

export const Sidebar: FC = () => {
  const { path, url } = useRouteMatch();
  const dispatch = useDispatch();

  return (
    <div className="page-container">
      <nav className="sidebar is-open">
        <div className="sidebar-header">
          <Logo />
        </div>
        <div className="sidebar-body">
          <ul className="sidebar-list">
            {sidebarTabsList.map((tab) => (
              <li className="sidebar-list-item" key={tab.href}>
                <NavLink
                  onClick={() => dispatch(setHeader([]))}
                  to={`${url}${tab.href}`}
                  role={tab.role || ''}
                  data-bs-toggle={tab.dataBsToggle || ''}
                  className="sidebar-link"
                  href={tab.href}
                  data-tab={tab.dataTab || ''}
                >
                  <div className="sidebar-icon">
                    <span className={tab.iconClass} />
                  </div>
                  <span className="sidebar-link-title">{tab.title}</span>
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
      </nav>

      <main className="main">
        <Switch>
          {sidebarTabsList.map((tab) => (
            <PrivateRoute
              exact
              path={`${path}${tab.href}`}
              component={() => <PageContentLayout title={tab.title}>{tab.component({})}</PageContentLayout>}
              key={tab.href}
            />
          ))}
          <PrivateRoute
            exact
            path={CLIENT_PATHS.security}
            component={() => (
              <PageContentLayout title="Security Development">
                <SecurityPage />
              </PageContentLayout>
            )}
          />
        </Switch>
      </main>
    </div>
  );
};
