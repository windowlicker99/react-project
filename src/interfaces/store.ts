import { ETableTypes, IColumn, ICurrentElement, IElementsTabsColumnsList, IUser } from '@interfaces/interfaces';

export interface IAuth {
  user: IUser;
}
export interface IPagination {
  nextPage: () => void;
  prevPage: () => void;
  pickPage: (indexPage: number) => void;
  disablePrev: boolean;
  disableNext: boolean;
  currentPage: number;
  pageAmount: number;
  currentTab: string;
  totalElement: number;
  totalElementOnPage: number;
  showedElements: number;
}

export interface IVehicleStore {
  currentVehicle: string;
}

export interface IHeaderStore {
  subtitles: string[];
}

export interface IChangeTypeAction {
  type: ETableTypes;
}

export interface IUpdateDisplayedColumnsAction {
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnState {
  type: ETableTypes;
  columns: IElementsTabsColumnsList;
}

export interface IStoreState {
  auth: IAuth;
  pagination: IPagination;
  vehicles: IVehicleStore;
  customizeColumns: ICustomizeColumnState;
  header: IHeaderStore;
  element: IElementStore;
}

export interface IDisplayColumnsList {
  [key: string]: boolean;
}

export interface ICustomizeColumnAction {
  type: ETableTypes;
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnsState {
  [key: string]: IColumn[];
}

export interface IElementStore {
  currentElement: ICurrentElement;
}
