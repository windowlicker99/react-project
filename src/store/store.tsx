import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import { authReducer } from '@store/slices/authSlice';
import { columnsReducer } from '@store/slices/columnsSlice';
import { paginationReducer } from '@store/slices/paginationSlice';
import { vehiclesReducer } from '@store/slices/vehicleSlice';
import { headerReducer } from '@store/slices/headerSlice';
import { elementReducer } from '@store/slices/elementSlice';

const reducers = combineReducers({
  auth: authReducer,
  vehicles: vehiclesReducer,
  customizeColumns: columnsReducer,
  pagination: paginationReducer,
  header: headerReducer,
  element: elementReducer,
});

const persistConfig = {
  key: 'root',
  storage,
  whiteList: ['auth', 'customizeColumns'],
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({
      serializableCheck: false,
    }),
  ],
});

export const persistor = persistStore(store);
