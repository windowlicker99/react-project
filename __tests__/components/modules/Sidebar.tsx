import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { store } from '@store/store';
import { Sidebar } from '@components/modules/Sidebar';

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as Record<string, unknown>), // use actual for all non-hook parts
  useParams: () => ({
    companyId: 'company-id1',
    teamId: 'team-id1',
  }),
  useRouteMatch: () => ({ url: '/company/company-id1/team/team-id1' }),
}));

describe('Sidebar tests', () => {
  it('Should render', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Sidebar />
        </BrowserRouter>
      </Provider>
    );
    expect(container.getElementsByClassName('sidebar').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Sidebar />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
