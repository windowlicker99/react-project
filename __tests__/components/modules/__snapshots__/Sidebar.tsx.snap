// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-container"
  >
    <nav
      class="sidebar is-open"
    >
      <div
        class="sidebar-header"
      >
        <a
          class="logo"
          href="/"
        >
          <span
            class="icon-logo"
          />
        </a>
      </div>
      <div
        class="sidebar-body"
      >
        <ul
          class="sidebar-list"
        >
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle=""
              data-tab="vehicle"
              href="/company/company-id1/team/team-id1/vehicle"
              role=""
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-vehicle"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Vehicle Types
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab="architecture"
              href="/company/company-id1/team/team-id1/architecture"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-architecture"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Architecture
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab="elements"
              href="/company/company-id1/team/team-id1/elements"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-elements"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Elements
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab="suppliers"
              href="/company/company-id1/team/team-id1/suppliers"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-user"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Suppliers
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab="tickets"
              href="/company/company-id1/team/team-id1/tickets"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-tickets"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Tickets
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/services"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-services"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Services
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/threats"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-threats"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Threats
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/controls"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-controls"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Controls
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/certification"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-certification"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Certification
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/reports"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-reports"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Reports
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/documents"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-documents"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Documents
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/notifications"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-notifications"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Notifications
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/feeds"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-feeds"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                External Feeds
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/configurations"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-сonfigurations"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Configurations
              </span>
            </a>
          </li>
          <li
            class="sidebar-list-item"
          >
            <a
              class="sidebar-link"
              data-bs-toggle="collapse"
              data-tab=""
              href="/company/company-id1/team/team-id1/settings"
              role="button"
            >
              <div
                class="sidebar-icon"
              >
                <span
                  class="icon-settings"
                />
              </div>
              <span
                class="sidebar-link-title"
              >
                Settings
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main
      class="main"
    />
  </div>
</DocumentFragment>
`;
